src/main/java/com/mongodb/kafka/connect/source/schema/BsonValueToSchemaAndValue.java

```diff
@@ -113,19 +120,24 @@ public SchemaAndValue toSchemaAndValue(final Schema schema, final BsonValue bson
   }
 
   private SchemaAndValue numberToSchemaAndValue(final Schema schema, final BsonValue bsonValue) {
-    if (!bsonValue.isNumber()) {
-      throw unexpectedBsonValueType(schema.type(), bsonValue);
-    }
     Object value;
-    BsonNumber bsonNumber = bsonValue.asNumber();
-    if (bsonNumber.isInt32()) {
-      value = bsonNumber.intValue();
-    } else if (bsonNumber.isInt64()) {
-      value = bsonNumber.longValue();
-    } else if (bsonNumber.isDouble()) {
-      value = bsonNumber.doubleValue();
+    if (bsonValue.isNumber()) {
+      BsonNumber bsonNumber = bsonValue.asNumber();
+      if (bsonNumber.isInt32()) {
+        value = bsonNumber.intValue();
+      } else if (bsonNumber.isInt64()) {
+        value = bsonNumber.longValue();
+      } else if (bsonNumber.isDouble()) {
+        value = bsonNumber.doubleValue();
+      } else {
+        value = bsonNumber.decimal128Value().bigDecimalValue();
+      }
+    } else if (bsonValue.isTimestamp()) {
+      value = bsonValue.asTimestamp().getTime();
```

Comment:
Note that BsonTimestamp.getTime returns seconds since the epoch, whereas BsonDateTime returns milliseconds since the epoch.  Should those be normalized?

Reply:
Yes - good catch.

Reply:
Done